src/Number.php

```diff
@@ -247,17 +255,23 @@ private static function parseIntegerPart($number)
             return '-0';
         }
 
+        // Happy path performance optimization: number can be used as-is if it is within
+        // the platform's integer capabilities.
+        if ($number === (string) (int) $number) {
```

Comment:
I'm very puzzled about this: it seems like there's no visible performance impact, although it **should** be faster, since the algorithm below is `O(n)`, but in userland :thinking: 

My thought was to keep things as "simple integers" where possible in internals, but that's probably going to complicate the code by a lot.

Reply:
Verified that there actually **is** a performance impact: comment above can be dismissed.